lib/fs.js

```diff
@@ -228,41 +243,43 @@ fs.existsSync = function(path) {
 };
 
 fs.readFile = function(path, options, callback_) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
+  return new Promise((resolve, reject) => {
+    var callback = maybeCallback(arguments[arguments.length - 1], resolve, reject);
```

Comment:
Hmm, what's your thinking of using `arguments[arguments.length - 1]` over `callback_`?